cozy/view_model/search_view_model.py

```diff
@@ -1,75 +1,74 @@
-import cozy.ext.inject as inject
+from typing import Callable
 
-from cozy.extensions.set import split_strings_to_set
-from cozy.open_view import OpenView
+from gi.repository import GLib
+
+import cozy.ext.inject as inject
 from cozy.application_settings import ApplicationSettings
 from cozy.architecture.event_sender import EventSender
 from cozy.architecture.observable import Observable
 from cozy.control.filesystem_monitor import FilesystemMonitor
+from cozy.extensions.set import split_strings_to_set
 from cozy.model.book import Book
 from cozy.model.library import Library
+from cozy.open_view import OpenView
 
 
 class SearchViewModel(Observable, EventSender):
     _fs_monitor: FilesystemMonitor = inject.attr("FilesystemMonitor")
     _model: Library = inject.attr(Library)
     _application_settings: ApplicationSettings = inject.attr(ApplicationSettings)
 
-    _search_open: bool = False
-
     def __init__(self):
         super().__init__()
         super(Observable, self).__init__()
 
-    @property
-    def books(self):
-        return self._model.books
-
-    @property
-    def authors(self):
+    def _get_available_books(self) -> set[Book]:
         is_book_online = self._fs_monitor.get_book_online
         show_offline_books = not self._application_settings.hide_offline
 
-        authors = {
-            book.author
-            for book
-            in self._model.books
-            if is_book_online(book) or show_offline_books
+        return {
+            book for book in self._model.books if is_book_online(book) or show_offline_books
```

Comment:
`is_book_online()` check is done even when `show_offline_books` is True and we could skip the check altogether?